toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -239,10 +239,56 @@ func CreateSparseDisk(diskPath string, size uint64, perm os.FileMode) (err error
 	return
 }
 
+// devineBlockSize tries to determine the block size used by the file system image
+func devineBlockSize(diskFilePath string) uint32 {
+	var blockSize uint32 = 512 // Default to 512
+	diskFile, err := os.Open(diskFilePath)
+	if err != nil {
+		logger.Log.Warnf("Failed to open %v to devine block size, assuming %d.", diskFilePath, blockSize)
+		return blockSize
+	}
+	defer diskFile.Close()
+
+	var probePoint int64 = 4096
+	signature := make([]byte, 8)
+	_, err = diskFile.ReadAt(signature, probePoint)
+	if err != nil {
+		logger.Log.Warnf("Failed to read from %v to devine block size, assuming %d.", diskFilePath, blockSize)
+		return blockSize
+	}
+	// GPT header starts with 8 bytes of ascii "EFI PART"
+	if signature[0] == 0x45 &&
+		signature[1] == 0x46 &&
+		signature[2] == 0x49 &&
+		signature[3] == 0x20 &&
+		signature[4] == 0x50 &&
+		signature[5] == 0x41 &&
+		signature[6] == 0x52 &&
+		signature[7] == 0x54 {
+		logger.Log.Debugf("Found GPT signature at %d in: %v, assuming that block size", probePoint, diskFilePath)
+		blockSize = uint32(probePoint)
+	}
+	return blockSize
+}
+
 // SetupLoopbackDevice creates a /dev/loop device for the given disk file
-func SetupLoopbackDevice(diskFilePath string) (devicePath string, err error) {
-	logger.Log.Debugf("Attaching Loopback: %v", diskFilePath)
-	stdout, stderr, err := shell.Execute("losetup", "--show", "-f", "-P", diskFilePath)
+func SetupLoopbackDevice(diskFilePath string, disk *configuration.Disk) (devicePath string, err error) {
```

Comment:
Pass in the `BlockSize` as a `uint64` (or `*uint64`) instead of taking in the entire `Disk` object. This will make the function easier to use in other contexts. e.g. the imager customizer tool.

Reply:
My idea here was to not have to de-encapsulate magic parameters out of the disk configuration before we get to the point where the parameters are consumed. If it is very objectionable, I guess we can pull out the block size farther up the call chain and pass it instead of (a pointer to) the configuration object.

Any manipulation of the disk image after it is partitioned, should use the same configuration as the disk image was created with.

In the case of the image customizer, I think the original configuration.Disk may not be available, so passing nil is really no worse than a hypothetical value for the block size that would represent do-the-right-thing-because-we-don't-know-the-block-size

Reply:
There is a good chance I'll need to add support for specifying the block size to the image customizer at some point. So, even if you don't fix this now, I may end up doing this in the future.

Reply:
I made the suggested change.